packages/dev/core/src/Materials/greasedLinePluginMaterial.ts

```diff
@@ -0,0 +1,668 @@
+    /**
+     * Creates a RawTexture from an RGBA color array and sets it on the plugin material instance.
+     * @param name name of the texture
+     * @param colors Uint8Array of colors
+     */
+    private _createColorsTexture(name: string, colors: Color3[]) {
+        const colorsArray = GreasedLinePluginMaterial._Color3toRGBAUint8(colors);
+        this._colorsTexture = new RawTexture(colorsArray, colors.length, 1, Engine.TEXTUREFORMAT_RGBA, this._scene, false, true, this._options.colorsSampling ?? RawTexture.NEAREST_NEAREST);
+        this._colorsTexture.name = name;
+    }
+
+    /**
+     * Disposes the plugin material.
+     */
+    public dispose(): void {
+        this._colorsTexture?.dispose();
+        super.dispose();
+    }
+
+    /**
+     * Sets whether to use the colors option to colorize the line.
+     * @param value true if use the colors, otherwise false
+     */
+    public setUseColors(value: boolean) {
+        this._options.useColors = value;
+    }
+
+    /**
+     * Creates or updates the colors texture
+     * @param colors color table RGBA
+     * @param lazy if lazy, the colors are not updated
+     * @param forceUpdate force creation of a new texture
+     * @returns
+     */
+    public setColors(colors: Nullable<Color3[]>, lazy = false, forceUpdate = false): void {
+        if (colors === null || colors.length === 0) {
+            this._colorsTexture?.dispose();
+            return;
+        }
+
+        const origColorsCount = this._options.colors?.length ?? 0;
+
+        this._options.colors = colors;
+
+        if (lazy && !forceUpdate) {
+            return;
+        }
+
+        if (this._colorsTexture && origColorsCount === colors.length && !forceUpdate) {
+            const colorArray = GreasedLinePluginMaterial._Color3toRGBAUint8(colors);
+            this._colorsTexture.update(colorArray);
+        } else {
+            this._colorsTexture?.dispose();
+            this._createColorsTexture(`${this._material.name}-colors-texture`, colors);
+        }
+    }
+
+    /**
+     * Updates the material. Use when material created in lazy mode.
+     */
+    public updateLazy() {
+        if (this._options.colors) {
+            this.setColors(this._options.colors, false, true);
+        }
+    }
+
+    /**
+     * Gets the plugin material options
+     * @returns the plugin material options @see GreasedLineMaterialOptions
+     */
+    public getOptions(): GreasedLineMaterialOptions {
+        return this._options;
+    }
+
+    /**
+     * Sets the line length visibility.
+     * 0 - 0% of the line will be visible
+     * 1 - 100% of the line will be visible
+     * @param value
+     */
+    public setVisibility(value: number) {
+        this._options.visibility = value;
+    }
+
+    /**
+     * Turns on/off dashmode
+     * @param value
+     */
+    public setUseDash(value: boolean) {
+        this._options.useDash = value;
+    }
+
+    /**
+     * Sets the dash array.
+     * @param value dash array

```

Comment:
I have renamed it back to setDashCount and also modified the dash calculation routine so now it is taking line segment length into account instead of point count so the dashes are now render correctly for any line and/or line segments. Basically, it just works.